src/writing-your-own-application/generate-project/index.md

```diff
@@ -0,0 +1,52 @@
+# Generating Projects from Templates
+
+We currently maintain two template repositories:
+- [esp-template] - `no_std` template.
+- [esp-idf-template] - `std` template.
+
+Both templates are based on [cargo-generate], a tool that allows you to create a new project based on some existing template. In our case [esp-idf-template] or [esp-template] can be used to generate an application with all the required configuration and dependencies.
+
+1. Install `cargo generate`:
+    ```shell
+    cargo install cargo-generate
+    ```
+2. Generate a project based in one of the templates:
+    - `std` template:
+        ```shell
+        cargo generate esp-rs/esp-idf-template cargo
+        ```
+        See [Understanding esp-idf-template] for more details on the template project.
+    - `no_std` template:
+        ```shell
+        cargo generate -a esp-rs/esp-template
+        ```
+        See [Understanding esp-template] for more details on the template project.
+
+    When the `cargo generate` subcommand is invoked, you will be prompted to answer a number of questions regarding the target of your application. Upon completion of this process you will have a buildable project with all the correct configuration.
+
+    The generated application can be built as normal using the appropriate toolchain and target simply by running `cargo build` when using either templates.
+
+    Using `cargo run` will compile the project, flash it, and open a serial monitor with our chip.
+
+## Using Dev Containers in the templates
```

Comment:
I see there are some instructions in esp-rs/esp-template. I skimmed through them, and they look serious.

As an idea, it might be great to have a tutorial about using Dev Containers in VS Code with the templates. Projects with Dev Containers are usually very good for beginners.

If I missed something or if it is irrelevant, please consider it as noise.

Reply:
By "too serious", I mean that it looks like normal documentation as it should of course. It has links to other resources that you might choose to go through and other things that you _can_ do.

A tutorial might be good from the perspective that you define a goal -- the result of the effort -- and lead the reader to this goal directly using only the instructions needed to achieve this goal without any additional information. The bottom line is the reader opening the document, following the steps without the need to make choices, and having a sample project completed by themselves with some desired result.

Again, like I said, it is just an idea off the top of my head which might be useful or might be completely irrelevant. You know the subject better, so you decide.